src/plugins/transponder/transponder_impl.cpp

```diff
@@ -17,9 +17,22 @@ TransponderImpl::~TransponderImpl()
     _parent->unregister_plugin(this);
 }
 
-void TransponderImpl::init() {}
+void TransponderImpl::init() 
+{
+    using namespace std::placeholders; // for `_1`
+
+    _parent->register_mavlink_message_handler(
+        MAVLINK_MSG_ID_ADSB_VEHICLE,
+        std::bind(&TransponderImpl::process_transponder, this, _1),
+        this);
+}
 
-void TransponderImpl::deinit() {}
+void TransponderImpl::deinit() 
+{
+    _parent->unregister_statustext_handler(this);
+    _parent->unregister_param_changed_handler(this);
```

Comment:
Why are those needed? They are not registered in the `init`, are they? :thinking:

Reply:
Ahh yeah you're right. I'll remove them. I was using the telemetry_imp.cpp as a guide and might have gotten carried away with what I was bring over. I've received the sensor however my test code seems to have issues. My subscription that I create should output text at 1 Hz however I never get any output from the function.

Is there another tie in somewhere that I need to get mavsdk to register my subscription? 

[Created Example](https://github.com/DoppleGangster/MAVSDK/blob/main/examples/transponder/transponder.cpp)